libr/flag/flag.c

```diff
@@ -1,29 +1,50 @@
 /* radare - LGPL - Copyright 2007-2016 - pancake */
 
+#include <r_cons.h>
 #include <r_flag.h>
+#include <r_types.h>
 #include <r_util.h>
-#include <r_cons.h>
 #include <stdio.h>
 
-R_LIB_VERSION(r_flag);
-
-/* aim to fix a bug in hashtable64 , collisions happen */
-/* offset needs to be xored to avoid some collisions !!! must switch to sdb */
-#define XORKEY 0x12345678
-#define XOROFF(x) (x^XORKEY)
+R_LIB_VERSION (r_flag);
 
 #define ISNULLSTR(x) (!(x) || !*(x))
 #define IS_IN_SPACE(f, i) ((f)->space_idx != -1 && (i)->space != (f)->space_idx)
 
+// Sources: http://burtleburtle.net/bob/hash/integer.html
+//          http://www.cris.com/~Ttwang/tech/inthash.htm
+// Described as public domain
+static ut32 hash_func_32(ut32 a) {
```

Comment:
I don't know all the detail but using XOR as hash func might be *why* you had lots of collisions.